assets/js/googlesitekit/datastore/site/enable-auto-update.js

```diff
@@ -0,0 +1,149 @@
+ *
+ * Site Kit by Google, Copyright 2022 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * External dependencies
+ */
+import invariant from 'invariant';
+
+/**
+ * Internal dependencies
+ */
+import Data from 'googlesitekit-data';
+import { CORE_SITE } from './constants';
+import { createFetchStore } from '../../data/create-fetch-store';
+import { actions as errorStoreActions } from '../../data/create-error-store';
+import { CORE_USER } from '../user/constants';
+
+const { receiveError, clearError } = errorStoreActions;
+const { createRegistrySelector } = Data;
+
+const fetchEnableAutoUpdateStore = createFetchStore( {
+	baseName: 'enableAutoUpdate',
+	controlCallback: async ( { nonce, pluginBasename } ) => {
+		const data = new FormData();
+		data.append( 'action', 'toggle-auto-updates' );
+		data.append( '_ajax_nonce', nonce );
+		data.append( 'state', 'enable' );
+		data.append( 'type', 'plugin' );
+		data.append( 'asset', pluginBasename );
+
+		const response = await fetch( global.ajaxurl, {
+			method: 'POST',
+			credentials: 'same-origin',
+			body: data,
+		} );
+
+		return response.json();
+	},
+	argsToParams: ( { nonce, pluginBasename } ) => {
+		return {
+			nonce,
+			pluginBasename,
+		};
+	},
+	validateParams: ( { nonce, pluginBasename } ) => {
+		invariant( typeof nonce === 'string', 'nonce must be a string.' );
+		invariant(
+			typeof pluginBasename === 'string',
+			'pluginBasename must be a string.'
+		);
+	},
+} );
+
+const baseInitialState = {};
+
+const baseActions = {
+	/**
+	 * Enables auto updates for Site Kit.
+	 *
+	 * @since n.e.x.t
+	 */
+	*enableAutoUpdate() {
+		yield clearError( 'enableAutoUpdate', [] );
+
+		const registry = yield Data.commonActions.getRegistry();
+
+		yield Data.commonActions.await(
+			registry.__experimentalResolveSelect( CORE_USER ).getNonces()
+		);
+		yield Data.commonActions.await(
+			registry.__experimentalResolveSelect( CORE_SITE ).getSiteInfo()
+		);
+
+		const nonce = registry.select( CORE_USER ).getNonce( 'updates' );
+		const pluginBasename = registry.select( CORE_SITE ).getPluginBasename();
+
+		const { response, error } =
+			yield fetchEnableAutoUpdateStore.actions.fetchEnableAutoUpdate( {
+				nonce,
+				pluginBasename,
+			} );
+
+		if ( response?.success ) {
+			yield registry
+				.dispatch( CORE_SITE )
+				.setSiteKitAutoUpdatesEnabled( true );
+		}
+
+		if ( error ) {
+			yield receiveError( error, 'enableAutoUpdate', [] );
+		}
```

Comment:
Is this necessary? This should happen automatically in the fetch store 🤔 Same with the `clearError` above.

Reply:
I'm not sure if this is necessary, but its consistent to other fetch store actions like [`setTrackingEnabled`](https://github.com/google/site-kit-wp/blob/68f0aebb9ffd0ab631982da846737663cd4141a5/assets/js/googlesitekit/datastore/user/tracking.js#L67-L79), [`saveUserInputSettings`](https://github.com/google/site-kit-wp/blob/68f0aebb9ffd0ab631982da846737663cd4141a5/assets/js/googlesitekit/datastore/user/user-input-settings.js#L150-L180), [`getProperties`](https://github.com/google/site-kit-wp/blob/68f0aebb9ffd0ab631982da846737663cd4141a5/assets/js/modules/analytics/datastore/properties.js#L413-L465) and some others.